source/h1_connection.c

```diff
@@ -1035,8 +1097,20 @@ static int s_decoder_on_request(
     void *user_data) {
 
     struct h1_connection *connection = user_data;
+
+    AWS_ASSERT(!connection->thread_data.incoming_stream);
+    /*make a new stream for this request and push it into the stream list wait for response
+      only for server side!*/
+    struct h1_stream *stream = s_new_server_stream(connection);
+    AWS_ASSERT(stream);
+    connection->base.server_data->on_incoming_request(
+        &connection->base, &stream->base, connection->base.server_data->connection_user_data);
+    connection->thread_data.incoming_stream = stream;
```

Comment:
We should check to ensure that the user called configure(), and handle the errors if they didn't